build-tools/packages/build-cli/src/commands/modify/fluid-imports.ts

```diff
@@ -0,0 +1,381 @@
+ */
+function getApiLevelForImportName(
+	name: string,
+	data: MemberData,
+	defaultValue: ApiLevel,
+	onlyInternal: boolean,
+): ApiLevel {
+	if (data.alpha?.includes(name) === true) return onlyInternal ? internalLevel : alphaLevel;
+	if (data.beta?.includes(name) === true) return onlyInternal ? internalLevel : alphaLevel;
+	if (data.public?.includes(name) === true) return publicLevel;
+	if (data.internal?.includes(name) === true) return internalLevel;
+	return defaultValue;
+}
+
+async function updateImports(
+	tsConfigFilePath: string,
+	mappingData: MapData,
+	onlyInternal: boolean,
+	organizeImports: boolean,
+	log?: CommandLogger,
+): Promise<void> {
+	const project = new Project({
+		tsConfigFilePath,
+	});
+	const sourceFiles = project
+		.getSourceFiles()
+		// Filter out type files - this may not be correct in projects with manually defined declarations.
+		.filter((sourceFile) => sourceFile.getExtension() !== ".d.ts");
+
+	/**
+	 * List of source file save promises. Used to collect modified source file save promises so we can await them all at
+	 * once.
+	 */
+	const fileSavePromises: Promise<void>[] = [];
+
+	// Iterate over each source file, looking for Fluid imports
+	for (const sourceFile of sourceFiles) {
+		log?.verbose(`Source file: ${sourceFile.getBaseName()}`);
+
+		/**
+		 * All of the import declarations. This is basically every `import foo from bar` statement in the file.
+		 */
+		let imports = sourceFile.getImportDeclarations();
+
+		// Skip source files with no imports.
+		if (imports.length === 0) {
+			continue;
+		}
+
+		/**
+		 * True if the sourceFile has changed.
+		 */
+		let sourceFileChanged = false;
+
+		/**
+		 * We'll populate the maps defined below this as we scan the existing imports, then write new remapped imports to
+		 * the file.
+		 */
+
+		/**
+		 * A mapping of new module specifier to named import. This map only contains "regular" imports; that it, it excludes
+		 * type-only imports.
+		 */
+		const newRegularImports: Map<string, Set<string>> = new Map();
+
+		/**
+		 * A mapping of new module specifier to named import. This map only contains type-only imports.
+		 */
+		const newTypeOnlyImports: Map<string, Set<string>> = new Map();
+
+		// FIRST PASS: collect the existing declarations
+		for (const importDeclaration of imports) {
+			// Skip non-Fluid imports
+			if (!isFluidImport(importDeclaration)) {
+				continue;
+			}
+			const [moduleName] = parseImport(importDeclaration);
+			const data = mappingData.get(moduleName);
+
+			// Skip modules with no mapping
+			if (data === undefined) {
+				log?.verbose(`Skipping ${importDeclaration.getModuleSpecifierValue()}`);
+			} else {
+				// TODO: Handle default import.
+				const defaultImport = importDeclaration.getDefaultImport();
+				if (defaultImport !== undefined) {
+					log?.warning(
+						`Found a default import (not yet implemented): ${defaultImport.getText().trim()}`,
+					);
```

Comment:
1. Warning is good, but this should also probably `continue` to avoid mangling.
2. log appears optional - should there be something stronger?